### PR TITLE
Cache paths and has_battery

### LINKED ISSUE #168: We check get_best_path() each time functions are called
These functions all call get_best_path every time they are called 
- read_power_source
- read_battery_charge
- read_lid_state

We should do something at the start to determine each path

### PR DESCRIPTION
**Pull request by JakeRoggenbuck**
fixes #168

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -189,6 +189,14 @@ fn parse_args(config: config::Config) {
         testing: false,
     };
 
+    let mut device_power = DevicePower {
+        did_init: false,
+        _has_battery: false,
+        _best_lid_path: String::new(),
+        _best_power_source_path: String::new(),
+        _best_battery_charge_path: String::new(),
+    };
```
Comment (Camerooooon) [inline src/main.rs:198]: DevicePower struct instantiation is duplicated here. I recommend using the Default trait.

**Review by Camerooooon — CHANGES_REQUESTED** [review]
One small issue

**Comment by Camerooooon** [comment]
For some reason when I run `cargo run monit` it does not work. Something is not working. There is no error message.

**Comment by JakeRoggenbuck** [comment]
i broke so much :sob: hold on i think I know what i did

**Comment by JakeRoggenbuck** [comment]
It might be worth just doing a rewrite now with the new interface

**Comment by JakeRoggenbuck** [comment]
I want to keep the idea of storing those paths in an object as to not recheck each time, but the other things I did aren't as important

**Pull request closed without merging**